services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/FileScanService.kt

```diff
@@ -0,0 +1,89 @@
+package fi.hel.haitaton.hanke.security
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import fi.hel.haitaton.hanke.toJsonString
+import java.time.Duration.ofSeconds
+import mu.KotlinLogging
+import org.apache.commons.lang3.BooleanUtils.isNotFalse
+import org.springframework.core.io.ByteArrayResource
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.MediaType.APPLICATION_PDF
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.BodyInserters.fromMultipartData
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+
+private val logger = KotlinLogging.logger {}
+
+private const val FORM_KEY = "FILES"
+
+@Service
+class FileScanService(
+    private val fileScanClient: WebClient,
+) {
+
+    fun scanFiles(files: Set<Pair<String, ByteArray>>): FileScanResult {
+        logger.info { "Scanning ${files.size} files." }
+        val response = getResults(files).also { it.validateStatus() }
+
+        val virusDetected = response.data.result.any { isNotFalse(it.isInfected) }
+
+        return FileScanResult(virusDetected, response).also { it.logStatus() }
+    }
+
+    private fun getResults(files: Set<Pair<String, ByteArray>>): FileScanResponse {
+        val data =
+            MultipartBodyBuilder()
+                .apply {
+                    files.forEach { (name, bytes) ->
+                        part(FORM_KEY, ByteArrayResource(bytes), APPLICATION_PDF).filename(name)
```

Comment:
The attachments are not all PDFs.

Reply:
Alright, would it be ok to define them as a parameter in scanFiles function?

Reply:
Aren't we now at the point where we need to measure the scan speed?

Reply:
https://helsinkisolutionoffice.atlassian.net/browse/HAI-1535 it's here.